### PR TITLE
Updated documentation (34365)

### PR DESCRIPTION
**Pull request by wener-tiobe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,31 +93,55 @@ jobs:
...
+| `norecalc`             | Comma-separated list of [metrics](https://portal.tiobe.com/latest/docs/index.html#doc=user/clientoptions.html%23MetricAliases) to not be recalculated. | -       |
+| `branchdir`            | Root directory of the source files for the branch.                                                                                                     | -       |
+| `branchname`           | Name of the branch in TICS.                                                                                                                            | -       |
+| `excludeMovedFiles`    | Exclude moved and renamed files from analysis completely. By default these are included if there are modifications in the file.                        | `false` |
+| `showBlockingAfter`    | Show the blocking after violations in the changed files window. Options are `true` or `false`.                                                         | `true`  |
+| `tmpdir`               | Location to store debug information.                                                                                                                   | -       |
+
+## Other features
```
Comment (janssen-tiobe) [inline README.md:132]: I would remove `Mode(s)` from this table as it is always for all modes.

File: README.md [hunk 2]
```diff
@@ -38,15 +39,41 @@ jobs:
...
+| `viewerUrl`     | A URL pointing to the "cfg" API endpoint of the TICS Viewer. It contains the name of the TICS Analyzer Configuration or "-" in case of the default configuration.                                                | true     |
+| `project`       | Name of the TICS project present in the TICS Viewer. If not given it will use project `auto` when running Client. Is required for QServer.                                                                       | false    |
+| `ticsAuthToken` | Authentication token to authorize the plugin when it connects to the TICS Viewer. (Only required if a token is needed to run TICS.)                                                                              | false    |
+| `installTics`   | Boolean parameter to install TICS command-line tools on a runner before executing the analysis. If not specified, TICS should be installed manually on the machine that runs this job, default value is `false`. | false    |
+
+### Advanced parameters
```
Comment (janssen-tiobe) [inline README.md:65]: I would add `nocalc` and `norecalc` in this table too, as it is used if set.

**Review by janssen-tiobe — CHANGES_REQUESTED** [review]
Like the changes overall. Have two little change requests.

File: README.md [hunk 3]
```diff
@@ -66,9 +101,55 @@ jobs:
...
+| `trustStrategy`        | Check the validity of certificates. Options are `all`, `self-signed` or `strict`. [Documentation on Client-side SSL/TLS](https://portal.tiobe.com/latest/docs/#doc=admin/admin_11_viewer.html%23ssl-wrapper).                                                                            | `strict`                                 |
+| `retryCodes`           | Status codes to retry api calls for. The default codes will be overwritten if this option is set.                                                                                                                                                                                        | `419`, `500`, `501`, `502`, `503`, `504` |
+| `secretsFilter`        | Comma-seperated list of extra secrets to mask in the console output.                                                                                                                                                                                                                     | -                                        |
```
Comment (zon-tiobe) [inline README.md]: Comma-seperated -> Comma-separated

File: README.md [hunk 4]
```diff
@@ -86,65 +167,23 @@ jobs:
...
-| `tmpdir`               | Location to store debug information.                                                                                                                                                                                                                                                     | -                                        | all             |
-| `displayUrl`           | The publicly available Viewer URL of TICS viewer to link the links in the review to. (e.g. https://domain.com/tiobeweb/TICS)                                                                                                                                                             | -                                        | all             |
+## Migration from v2 to v3
+The interface (running with params) have been changed to depict their usage better or to synchronise with the TICS command line usage and by synchronous with other plugins we provide.
```
Comment (zon-tiobe) [inline README.md]: Sentence doesn't flow that nicely. I'd use something like this:

The interface (running with parameters) has been changed to better reflect its usage. This interface is now more synchronized with TICS command line usage, and have been changed to be closer to other plugins that TIOBE provides.

File: README.md [hunk 5]
```diff
@@ -66,9 +101,55 @@ jobs:
...
+| `secretsFilter`        | Comma-seperated list of extra secrets to mask in the console output.                                                                                                                                                                                                                     | -                                        |
+| `displayUrl`           | The URL that end-users need to visit the TICS Viewer. This can differ from the viewerURL if TICS Client and TICSQServer communicate via a reverse proxy, but the end-users have direct access. (e.g. https://domain.com/tiobeweb/TICS)             | -                                        |
+
+### Diagnostic Analysis mode
+
+There is also the possibility to do a so called "diagnostic" run. This mode can be enabled to test if TICS has been setup properly and can run on the machine the action is run on.
```
Comment (zon-tiobe) [inline README.md]: setup -> set up

File: README.md [hunk 6]
```diff
@@ -66,9 +101,55 @@ jobs:
...
+| `postAnnotations`      | Show the TICS violations in the changed files window. Options are `true` or `false`.                                                                                                                                                                                                     | `true`                                   |
+| `postToConversation`   | Post the summary to the conversation page of the pull request.                                                                                                                                                                                                                           | `true`                                   |
+| `pullRequestApproval`  | Set the plugin to approve or deny a pull request, by default this is false. Options are `true` or `false`. Note that once a run that added a reviewer has been completed, this reviewer cannot be deleted from that pull request. (Always the case on versions between 2.0.0 and 2.5.0). | `false`                                  |
```
Comment (zon-tiobe) [inline README.md]: Version of which tool? It helps to be certain about that :)

File: README.md [hunk 7]
```diff
@@ -38,15 +45,43 @@ jobs:
...
+| `norecalc`             | Comma-separated list of [metrics](https://portal.tiobe.com/latest/docs/index.html#doc=user/clientoptions.html%23MetricAliases) to not be recalculated. | -           |
+| `filelist`             | Path to a file containing the files (newline separated) to run TICS for. This can be an absolute or relative (to workspace) path, and can also be `.` to analyze the whole project. This has to be set when the action is run outside of a pull request. | -       |
+| `cdtoken`              | A custom client-data token for the purpose of the Client Viewer functionality. This provides a static URL that is updated with every analysis.         | -           |
+| `branchname`           | Name of the branch in TICS.                                                                                                                            | -           |
+| `codetype`             | Allows you to pick which specific types of code you want to analyze with the TICS client. Options are `PRODUCTION`, `TESTCODE` and `EXTERNAL`.         | `PRODUCTION`|
```
Comment (zon-tiobe) [inline README.md]: Technically, 'GENERATED' is also an option now. But that's probably the least interesting one.

File: README.md [hunk 8]
```diff
@@ -4,28 +4,35 @@
 [![Tests](https://github.com/tiobe/tics-github-action/actions/workflows/test.yml/badge.svg)](https://github.com/tiobe/tics-github-action/actions/workflows/test.yml)
 [![CodeQL](https://github.com/tiobe/tics-github-action/actions/workflows/codeql.yml/badge.svg)](https://github.com/tiobe/tics-github-action/actions/workflows/codeql.yml)
 
-The TICS Github action integrates TICS Client analysis to measure your code quality. The incorporated Quality gating feature enables you to analyze and decorate pull requests.
+The TICS Github action integrates the [TICS code quality framework](https://www.tiobe.com/products/tics/) into your workflows. This allows you to effectively measure and monitor the software code quality of all your projects and its branches.
```
Comment (brouwers-tiobe) [inline README.md]:
```suggestion
The TICS GitHub action integrates the [TICS code quality framework](https://www.tiobe.com/products/tics/) into your workflows. This allows you to effectively measure and monitor the software code quality of all your projects and their branches.

```

File: README.md [hunk 9]
```diff
@@ -4,28 +4,35 @@
...
 
-The TICS Github action integrates TICS Client analysis to measure your code quality. The incorporated Quality gating feature enables you to analyze and decorate pull requests.
+The TICS Github action integrates the [TICS code quality framework](https://www.tiobe.com/products/tics/) into your workflows. This allows you to effectively measure and monitor the software code quality of all your projects and its branches.
+The incorporated `Quality Gating` feature can be used for `Pull Request` and `Commit` approvals. It can also decorate your pull request so the quality data is easily available. 
+Furthermore, the changed files are annotated with findings from the TICS analysis, so it is clear where the issues are and thus where they need to be addressed.
 
-## Before you start
+There are two types of analysis modes available:
+- Reference runs using TICS QServer (See section QServer). Reference points with code quality metric data are created that are used for the qualification runs. Intended for branches where other branches are derived from and will be merged back to (Like main).
```
Comment (brouwers-tiobe) [inline README.md]: > Intended for branches where other branches are derived from and will be merged back to (Like main).

This reads a bit confusing to me. Perhaps we can incorporate the term 'base branch', which seems to be the conventional name in GitHub for 'the branch PRs are merged to'. For example: 'Intended for base branches of pull requests, e.g. `main`.' Similarly, we could refer to the 'branch to be merged' as 'head branch' of a PR

File: README.md [hunk 10]
```diff
@@ -4,28 +4,35 @@
...
+- Qualification runs using TICS Client (See section Client). Compares a set of changed files from a commit or pull request to the reference point for qualification, using a Quality Gate. Intended for any branch that needs to be merged back.
 
+## Before you start
 ### Prerequisites
+A TICS Viewer (version 2022.4 or higher) running somewhere on the network that is HTTP(S) accessible by the runner on which you want to execute the action.
 
-- A TICS Viewer (version 2022.4 or higher) running somewhere on the network that is HTTP(S) accessible by the runner on which you want to execute the action.
+### Supported Platforms
+Linux and Windows based runners, both Github-hosted and self-hosted, are supported.
```
Comment (brouwers-tiobe) [inline README.md]:
```suggestion
Linux and Windows based runners, both GitHub-hosted and self-hosted, are supported.

```
(nitpick)

File: README.md [hunk 11]
```diff
@@ -38,15 +45,43 @@ jobs:
...
+| `installTics`   | Boolean parameter to install TICS command-line tools on a runner before executing the analysis. If not specified, TICS should be installed manually on the machine that runs this job, default value is `false`. | false    |
+
+### Advanced parameters
+The following options allow to instrument TICS Client more specifically:
+
+| Input                  | Description                                                                                                                                            | Default     |
+|------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
+| `calc`                 | Comma-separated list of [metrics](https://portal.tiobe.com/latest/docs/index.html#doc=user/clientoptions.html%23MetricAliases) to be used. The `GATE` metric is supported for TICS Viewer versions higher than 2022.2.x.              | `GATE`  |
```
Comment (brouwers-tiobe) [inline README.md]:
```suggestion
| `calc`                 | Comma-separated list of [metrics](https://portal.tiobe.com/latest/docs/index.html#doc=user/clientoptions.html%23MetricAliases) to be calculated. The `GATE` metric is supported for TICS Viewer versions higher than 2022.2.x.              | `GATE`  |

```
(To keep in line with `nocalc`)

File: README.md [hunk 12]
```diff
@@ -66,9 +101,55 @@ jobs:
...
+| `branchdir`            | Root directory of the source files for the branch.                                                                                                     | -       |
+| `branchname`           | Name of the branch in TICS.                                                                                                                            | -       |
+| `excludeMovedFiles`    | Exclude moved and renamed files from analysis completely. By default these are included if there are modifications in the file.                        | `false` |
+| `showBlockingAfter`    | Show the blocking after violations in the changed files window. Options are `true` or `false`.                                                         | `true`  |
+| `tmpdir`               | Location to store debug information.                                                                                                                   | -       |
+
+## Other features
+### Action parameters
+Below are some special parameters that can be used to control how the Github Action posts its results:
```
Comment (brouwers-tiobe) [inline README.md]:
```suggestion
Below are some special parameters that can be used to control how the GitHub Action posts its results:

```

**Review by janssen-tiobe — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -86,65 +170,23 @@ jobs:
...
-| `tmpdir`               | Location to store debug information.                                                                                                                                                                                                                                                     | -                                        | all             |
-| `displayUrl`           | The publicly available Viewer URL of TICS viewer to link the links in the review to. (e.g. https://domain.com/tiobeweb/TICS)                                                                                                                                                             | -                                        | all             |
+## Migration from v2 to v3
+The interface (running with parameters) has been changed to better reflect its usage. This interface is now more synchronized with TICS command line usage, and have been changed to be closer to other plugins that TIOBE provides.
```
Comment (zon-tiobe) [inline README.md]: and have been changed -> and has been changed

File: README.md [hunk 14]
```diff
@@ -38,15 +47,43 @@ jobs:
...
+| `branchname`           | Name of the branch in TICS.                                                                                                                                 | -           |
+| `codetype`             | Allows you to pick which specific types of code you want to analyze with the TICS client. Options are `PRODUCTION`, `TESTCODE`, `EXTERNAL` and `GENERATED`. | `PRODUCTION`|
+| `excludeMovedFiles`    | Exclude moved and renamed files from analysis completely. By default these are included if there are modifications in the file.                             | `false`     |
+| `showBlockingAfter`    | Show the blocking after violations in the changed files window. Options are `true` or `false`.                                                              | `true`      |
+| `tmpdir`               | Location to store debug information.                                                                                                                        | -           |
+
+## QServer
+
+As of v3, the option to run [TICSQServer](https://ticsdocumentation.tiobe.com/latest/docs/#doc=admin/admin_A3_qserverref.html) analyses has been made available.
```
Comment (zon-tiobe) [inline README.md:84]: We are pointing to two different documentation links in the README:

https://portal.tiobe.com/latest/docs/index.html#doc=user/clientoptions.html%23MetricAliases
https://ticsdocumentation.tiobe.com/latest/docs/#doc=admin/admin_A3_qserverref.html

We should use one or the other in the entire README.

**Review by zon-tiobe — CHANGES_REQUESTED** [review]

**Review by zon-tiobe — APPROVED** [review]
LGTM!

**Pull request merged**